### PR TITLE
Prefetch While Computing

### PR DESCRIPTION
**Pull request by KarlLeell**

### CODE REVIEW

File: src/storage/gpu_swap_prefetch.cc [hunk 1]
```diff
@@ -64,19 +81,51 @@ void Prefetch::StopPrefetching() {
...
+      Swap::Get()->GetAddr(r.handle_id, true);
+    } else {
+      std::cout << "non-read operation found" << std::endl;
+    }
   }
+
 }
 
+
 // TODO(sotskin): karll: Add algorithm discription
```
Comment (Sotskin) [inline src/storage/gpu_swap_prefetch.cc:132]: Add algorithm discription

File: src/storage/gpu_swap_prefetch.cc [hunk 2]
```diff
@@ -64,19 +81,51 @@ void Prefetch::StopPrefetching() {
...
 
 
 void Prefetch::Prefetching(int device) {
   while(!stop_prefetching_) {
     (this->*DoPrefetch)(device);
     start_prefetching_ = true;
+  }
+}
+
+
```
Comment (Sotskin) [inline src/storage/gpu_swap_prefetch.cc:99]: Add algorithm description

**Review by Sotskin — CHANGES_REQUESTED** [review]
Looks good. Please add some simple descriptions for the algorithm

**Review by fegin — APPROVED** [review]
You may have a conflict (benchmark/run.sh) when merging this pull request because of @Sotskin 's pull request. Make sure you merge this pull request after he merges and resolves the conflict if there is one.

**Pull request merged**